pkgs/by-name/as/asset-ripper/package.nix

```diff
@@ -0,0 +1,65 @@
+{
+  autoPatchelfHook,
+  buildDotnetModule,
+  dbus,
+  dotnetCorePackages,
+  fetchFromGitHub,
+  lib,
+}:
+
+buildDotnetModule (finalAttrs: {
+  pname = "AssetRipper";
+  version = "1.1.3";
+
+  src = fetchFromGitHub {
+    owner = "AssetRipper";
+    repo = "AssetRipper";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-KSCM4HY+IjbWnhfOiTxp3WEJPjeOgPo4r9p1CjvBshs=";
+  };
+
+  # Until https://github.com/AssetRipper/AssetRipper/issues/1414 is resolved.
+  patches = [ ./replace-ro-exe-dir-with-cwd.patch ];
+
+  buildInputs = [ dbus.lib ];
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  # Prevent automatic patching of all files. This is necessary as applying
+  # autoPatchelf indiscriminately causes dangling references to openssl and
+  # icu4c in AssetRipper.GUI.Free
+  dontAutoPatchelf = true;
+
+  # Patch some prebuilt libraries fetched via NuGet.
+  fixupPhase = ''
+    runHook preFixup
+
+    autoPatchelf $out/lib/AssetRipper/libnfd.so
+    autoPatchelf $out/lib/AssetRipper/libTexture2DDecoderNative.so
+
+    runHook postFixup
+  '';
+
+  projectFile = "Source/AssetRipper.GUI.Free/AssetRipper.GUI.Free.csproj";
+
+  # Error: "PublishTrimmed is implied by native compilation and cannot be disabled."
+  # We need to override the project settings and disable native AoT compilation
+  # as this is incompatible with PublishTrimmed.
+  dotnetInstallFlags = [ "-p:PublishAot=false" ];
+
+  nugetDeps = ./deps.nix;
+
+  executables = [ "AssetRipper.GUI.Free" ];
```

Comment:
This executable name is a bit too unconventional to be used as the name of the main program. Maybe use `ln -s` to create an executable at `$out/bin/AssetRipper` that links to the built executable instead?

Reply:
`assetripper-gui`? There's no CLI in upstream, but I have [a small patch](https://github.com/YoshiRulz/AssetRipper/commit/c6221ea0d73585eb538f850f805d68ee1558b5de.patch) to add one, was going to suggest that for Nixpkgs.

Reply:
I think you should open a PR to upstream instead. Packaging should not add new features, especially such a significant one.

Reply:
AssetRipper/AssetRipper#1565

Reply:
Is the new executable name intuitive? I can resolve this conversation if it is.

Reply:
I was thinking setting `executables = [];` and linking to `$out/lib/${pname}/AssetRipper.GUI.Free` instead. Also technically this step should go in install phase instead of fixup phase. Also `meta.mainProgram` needs changing.

Reply:
My rationale for leaving `executables` as is to allow for the fixup phase hook to wrap the binary here:
https://github.com/NixOS/nixpkgs/blob/a01ee2f7f3e0f33f6738b6f49d24d3653a415682/pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-fixup-hook.sh#L84

If I set `executables = [];`, the main program does get generated in `lib` but is unable find the dotnet runtime when run from there. I am pushing a new commit that shifts the symlink-ing to `postInstall` and changes mainProgram.
Is setting `postInstall` the correct practice?

Reply:
Ah sorry didn't know that you cannot just launch from `lib`. `postInstall` seems correct.